ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -49,6 +50,235 @@ uint64_t getNewTraceIdx() {
 // The name prefix used for blocks that are branched to when a guard succeeds.
 #define GUARD_SUCCESS_BLOCK_NAME "guardsuccess"
 
+// Describes a "resume point" in an LLVM IR basic block.
+//
+// This is required because in LLVM IR, a call instruction does not terminate a
+// basic block, and so when we encounter a call, we have to keep track of our
+// porgress through the block. Upon returning from the call, we will skip ahead
+// to the instruction *after* the resume point.
+struct BlockResumePoint {
+  // The index of the basic block (in the parent function) containing the
+  // instruction at which to resume.
+  size_t ResumeBBIdx;
+  // The instruction *after* which to resume.
+  //
+  // To be clear, if we have two instructions:
+  // ```
+  //   call @f()
+  //   %x = load(...)
+  // ```
+  //
+  // If we want to resume at the `load`, then we store the `call`.
+  Instruction *ResumeAfterInstr;
+  // The index of the instruction immediately *before* where we should resume.
+  //
+  // This could be computed by looping over `ResumeInstr`'s parent block, but
+  // since it's needed fairly frequently, we cache it.
+  size_t ResumeAfterInstrIdx;
+
+  // Dump the resume point to stderr for debugging.
+  void dump() {
```

Comment:
Can we remove this debugging code?

Reply:
Why? You wouldn't object to a `impl Debug` implementation in Rust. This is equivalent. It's all nicely encapsulated and we even follow the LLVM convention of naming the functions "dump".

(I grow tired of repeatedly writing (non-trivial) debugging routines every time I want to do some debugging. C++ isn't like rust, where you can easily derive `Debug` printing -- everything has to be done manually, and it's very laborious)